12 commit

VeronikaMads opened the pull request:
пока не доделала задание

VeronikaMarss committed:
12 commit
пока не доделала задание

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,120 @@
+         * Если подходит, то выводим на экран.
+         */
+
+//        1.substring(): возвращает подстроку, начиная с определенного индекса до конца или до определенного индекса
```
когда отправляете код на проверку не забывайте раскоментировать код который вы закоментировали

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,120 @@
+//        int length = str5.length();
+//        for (int i = length - 1; i >= 0; i--)
+//            reverse = reverse + str5.charAt(i);
+//        if (str5.equals(reverse))
```
скобочки забываете { } в конструкции блока if

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,120 @@
+//        String str5 = "довод";
+//        String reverse = "";
+//        int length = str5.length();
+//        for (int i = length - 1; i >= 0; i--)
```
скобочки забыли добавить в блоке for () { }

VeronikaMarss committed:
12 commit
пока не могу сдвинутся с места.
как разбить предложения на длину - проблема. метод не работает.
тоже самое искать полиндром в тексте.

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,142 @@
+            System.out.println(" Is a palindrome.");
+        } else
+            System.out.println("Is not a palindrome.");
+//        3.2 способ
```
} скобки забыли после else

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,142 @@
+            while (i > j) {
+                if (str4.charAt(i) != str4.charAt(j)) {
+                    isPalindrome = false;
+                    System.out.println("Entered string is not a palindrome.");
```
break; еще нужен. если слово не является полиндромом то зачем дальше продолжать бегать по циклу `while (i > j)`

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,142 @@
+                "По земле, по лугам, по оврагам плывёт звон.Что это звенит?Вот скатилась капля сладкого сока с берёзовой ветки.";
+//
+//        выводит количество предложений (хотела сдесь добавить фильтр и разбить на слова , но не поняла как)
+        String[] sentences = text.split("[\\.\\!\\?]");
```
если у нас стоят квадратные скобки то можно не ставить \\ просто пишем разделители [.!?]

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,142 @@
+                "Разлилась волна зелёного тумана по широкому лугу.Стоят тёплые и тихие вечера.Звенят луга." +
+                "По земле, по лугам, по оврагам плывёт звон.Что это звенит?Вот скатилась капля сладкого сока с берёзовой ветки.";
+//
+//        выводит количество предложений (хотела сдесь добавить фильтр и разбить на слова , но не поняла как)
```
```
public static void main(String[] args) {
        final String text = "Пробудилась земля от долгого зимнего сна.Заблестела молодая травка." +
                "Разлилась волна зелёного тумана по широкому лугу.Стоят тёплые и тихие вечера.Звенят луга." +
                "По земле, по лугам, по оврагам плывёт звон.Что это звенит?Вот скатилась казак капля сладкого сока с берёзовой ветки.";
        String[] sentences = text.split("[.!?]");
        List<String> stringList = Stream.of(sentences)
                                        .map(s -> getSentenceByCondition(s))
                                        .filter(Objects::nonNull)
                                        .collect(Collectors.toList());
        System.out.println(stringList);
    }

    private static String getSentenceByCondition(String sentence) {
        String[] strings = sentence.split(" ");
        if (checkLength(strings) || checkPalindrome(strings)) {
            return sentence;
        }
        return null;
    }

    private static boolean checkPalindrome(String[] strings) {
        return Stream.of(strings)
                     .filter(s1 -> s1.length() > 1)
                     .anyMatch(s -> new StringBuilder(s).reverse().toString().equals(s));
    }

    private static boolean checkLength(String[] strings) {
        return strings.length >= 3 && strings.length <= 5;
    }

```

pavelTMS commented on homeWork12/src/main/java/p12/Main.java:
```diff
@@ -0,0 +1,142 @@
+        System.out.println();
+
+//          Дожно делить на слова, но не делит. Почему????
+        String[] word = text.split("[\\.\\!\\?]");
```
что бы разделить на слова надо указать другой делитель. это пробел.
text.split(" ");

pavelTMS reviewed (commented):
посмотрел.
